src/dataAccessLayer.ts

```diff
@@ -32,18 +35,46 @@ export class DataAccessLayer {
         return await this.blockchainRepository.getBlockchainAsync();
     }
 
+    public async getBlocksByPeriodAsync(beginTimestamp: number, endTimestamp: number): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByPeriodAsync(beginTimestamp, endTimestamp);
+    }
+
+    public async getBlocksByDateAsync(timestamp: number): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByDateAsync(timestamp);
+    }
+
+    public async getBlocksByHashAsync(blockHash: string): Promise<Block[]> {
+        return await this.blockchainRepository.getBlocksByHashAsync(blockHash);
+    }
+
     public async getBlockAsync(blockNumber: number): Promise<Block> {
         return await this.blockchainRepository.getBlockAsync(blockNumber);
     }
 
-    public async setBlocksAsync(blocks: Block[]): Promise<void> {
-        await this.blockchainRepository.setBlocksAsync(blocks);
+    public async getPreviousBlockAsync(parentHash: string): Promise<Block> {
+        return await this.blockchainRepository.getPreviousBlockAsync(parentHash);
+    }
+
+    public async getNextBlockAsync(blockHash: string): Promise<Block> {
+        return await this.blockchainRepository.getNextBlockAsync(blockHash);
+    }
+
+    public async createBlocksAsync(blocks: Block[]): Promise<void> {
+        await this.blockchainRepository.createBlocksAsync(blocks);
+    }
+
+    public async createBlockAsync(block: Block): Promise<void> {
+        await this.blockchainRepository.createBlockAsync(block);
     }
 
     public async deleteBlocksAsync(blockNumbers: number[]): Promise<void> {
```

Comment:
This is a fault in my initial implementation, but it would be better to rename the method too:
```ts
public async deleteBlocksByNumbersAsync(blockNumbers: number[])
```
> this would clearly indicate what parameter should be passed in the method.